### PR TITLE
Download the agent based on partition

### PR DESCRIPTION
**Pull request by richardpen**
### Testing

make test
New tests cover the changes: 
yes
### Licensing

This contribution is under the terms of the Apache 2.0 License: 
yes

### CODE REVIEW

File: ecs-init/config/common_test.go [hunk 1]
```diff
@@ -47,22 +48,38 @@ func TestDockerUnixSocketWithDockerHost(t *testing.T) {
...
+			err:         nil,
+		}, {
+			region:      "us-gov-west-1",
+			destination: "us-gov-west-1",
+		}, {
+			region:      "cn-north-1",
+			destination: "cn-north-1",
+		}, {
+			region:      "invalid",
+			destination: "",
```
Comment (jahkeup) [inline ecs-init/config/common_test.go]: I think we could also then use the err field to assert there should be an error for this testcase, or we could get rid of the field entirely.

File: ecs-init/config/common_test.go [hunk 2]
```diff
@@ -47,22 +48,38 @@ func TestDockerUnixSocketWithDockerHost(t *testing.T) {
...
+		destination string
+		err         error
 	}{
-		{DefaultRegionName, true},
-		{"cn-north-1", true},
-		{"missing-region", false},
+		{
+			region:      "us-west-2",
+			destination: "us-east-1",
+			err:         nil,
```
Comment (jahkeup) [inline ecs-init/config/common_test.go]: `err` isn't needed here.

File: ecs-init/config/common.go [hunk 3]
```diff
@@ -44,17 +47,25 @@ const (
 	DefaultRegionName = "us-east-1"
```
Comment (jahkeup) [inline ecs-init/config/common.go]: Is this unused now?

Comment (richardpen) [reply ecs-init/config/common.go]: It's still used as a default if for some unknown reason the region isn't mapped to any of the partitions.

File: ecs-init/config/common.go [hunk 4]
```diff
@@ -44,17 +47,25 @@ const (
...
 )
 
-// supportedRegions stores a list of regions that contains the agent bucket
-var supportedRegions = []string{"cn-north-1", "us-gov-west-1", DefaultRegionName}
+var partitionBucketMap = map[string]string{
+	"aws":        "us-east-1",
```
Comment (samuelkarp) [inline ecs-init/config/common.go]: You can use [these constants](https://github.com/aws/aws-sdk-go/blob/9a2fe34af9644afba4a1a1406966e78eb0e985af/aws/endpoints/defaults.go#L11-L13) and [these constants](https://github.com/aws/aws-sdk-go/blob/9a2fe34af9644afba4a1a1406966e78eb0e985af/aws/endpoints/defaults.go#L16-L44)

```go
var partitionBucketMap = map[string]string{
	endpoints.AwsPartitionID:      endpoints.UsEast1RegionID,
	endpoints.AwsCnPartitionID:    endpoints.CnNorth1RegionID,
	endpoints.AwsUsGovPartitionID: endpoints.UsGovWest1RegionID,
}
```

File: ecs-init/config/common.go (hunk 3 above)
Comment (samuelkarp) [inline ecs-init/config/common.go]: `endpoints.UsEast1RegionID`

**Pull request merged**